### PR TITLE
Improve i18n/a11y

### PR DESCRIPTION
**Pull request by apalaniuk**
* Change `<h3>` to `<p>` for name to provide a better screen reader experience (Need to test in legitimate instance to ensure this doesn't cause unintentional styling changes, but demo page with `.d2l-typography` set on body shows no change)
* Make user tile info (grade/school) RTL-friendly
* Add `d2l-demo-template` to provide a demo page that makes it easier to confirm RTL/etc.
* Add unit tests (not hooked up in CI yet, and fairly basic)
* Add a11ySuite tests (not that it can catch much at this level, but it _is_ useful)

### CODE REVIEW

File: test/d2l-user-tile/d2l-user-tile.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+				Polymer.dom.flush();
+			});
+
+			it('should render the custom icon only', function() {
+				expect(component.querySelector('d2l-icon')).to.not.exist;
+				expect(component.querySelector('d2l-image')).to.exist;
+			});
+		});
+	});
+});
```
Comment (apalaniuk) [inline test/d2l-user-tile/d2l-user-tile.js]: I tried for a while to get tests for tab interaction working to test that focus switches between tiles, but it seemed untestable at this level (following a similar pattern as https://github.com/PolymerElements/iron-overlay-behavior/blob/master/test/iron-overlay-behavior.html#L642). I'll revisit that later.

File: d2l-user-tile.html [hunk 2]
```diff
@@ -81,9 +90,11 @@
 				</template>
 			</div>
 			<div class="user-tile-information-wrapper">
-				<h3 class="user-tile-name">[[name]]</h3>
+				<p class="user-tile-name">[[name]]</p>
```
Comment (apalaniuk) [inline d2l-user-tile.html]: This was changed as screen readers would read out 'heading 3' in the middle of reading content, which is a bit jarring and not really necessary as I don't think it has actual heading semantics. The demo page sets base d2l-typography styles and there's no stylistic difference between the two that I can see.

File: test/d2l-user-tile/d2l-user-tile.js (hunk 1 above)
Comment (jstefaniuk-d2l) [inline test/d2l-user-tile/d2l-user-tile.js]: I've found only pain when trying to test tab flows

**Comment by jstefaniuk-d2l** [comment]
code LGTM :+1:

**Comment by JoshuaKlassen** [comment]
Testing notes:

---
Testing plan:

---
Tested on:
- [ ] Chrome

- [ ] FireFox

- [ ] Internet Explorer

- [ ] Edge

- [ ] Safari

- [ ] iOS

- [ ] Android

---
Other notes:

**Pull request merged**